### PR TITLE
Initial windows testing in CI support

### PR DESCRIPTION
**Pull request by peterwaller-arm**
This runs the `check.sh` under wine.

Some things are broken and need fixing. It would be interesting to know if they're broken under real windows as well or not.

For now I've suppressed the broken tests on windows so we can merge this and have a subsequent PR to re-enable the tests and fix them.

### CODE REVIEW

File: src/bin/elfshaker/store.rs [hunk 1]
```diff
@@ -102,6 +107,13 @@ fn find_files() -> Vec<PathBuf> {
     WalkDir::new(".")
         .follow_links(true)
         .into_iter()
+        .filter_entry(|p| {
+            p.path()
+                .components()
+                .nth(1)
+                .and_then(|p| Some(p.as_os_str() != Path::new("elfshaker_data")))
```
Comment (veselink1) [inline src/bin/elfshaker/store.rs:114]: Clippy warns here

File: src/repo/repository.rs [hunk 2]
```diff
@@ -356,7 +356,8 @@ impl Repository {
         let PackId::Pack(pack_name) = pack_id;
 
         // The pack is loose if the .pack.idx is in the loose packs directory
-        if !pack_name.starts_with(&(LOOSE_DIR.to_owned() + "/")) {
+        if !pack_name.starts_with(&(LOOSE_DIR.to_owned() + &std::path::MAIN_SEPARATOR.to_string()))
```
Comment (veselink1) [inline src/repo/repository.rs:359]: Clippy warning

File: test-scripts/check.sh [hunk 3]
```diff
@@ -48,7 +48,8 @@ all_pwd_sha1sums() {
 }
 
 elfshaker_sha1sums() {
-  "$elfshaker" list-files "$@" 2> /dev/null | awk '{print $1" "$3}' | sort -k2,2
+  # Note: Canonicalizes windows paths to unix paths for now.
+  "$elfshaker" list-files "$@" 2> /dev/null | awk '{print $1" "$3}' | sort -k2,2 | sed 's|\\|/|g'
```
Comment (veselink1) [inline test-scripts/check.sh:52]: (in the future) we could make changes to print paths in Unix-style, regardless of what the native separator is. Git certainly doesn't use the Windows path separator when running on Windows.

**Review by veselink1 — APPROVED** [review]
Change looks good. Clippy has warned at two places. I've not looked at the Nix files as I don't understand them :)

File: test-scripts/check.sh (hunk 3 above)
Comment (peterwaller-arm) [reply test-scripts/check.sh:52]: Sounds good to me (for the future :))

**Comment by peterwaller-arm** [comment]
Thanks for taking a look and spotting the clippy complaints. Fixed. Will merge when green.

**Pull request merged**